pcdsdevices/pseudopos.py

```diff
@@ -459,3 +460,97 @@ def inverse(self, real_pos: tuple) -> tuple:
             self._table_data_by_name[pseudo_field]
         )
         return self.PseudoPosition(**{pseudo_field: pseudo_value})
+
+
+class OffsetMotorBase(FltMvInterface, PseudoPositioner):
+    """
+    Motor with an offset.
+    """
+    motor = None
+    pseudo_motor = Cpt(PseudoSingleInterface, kind='normal')
+    user_offset = Cpt(EpicsSignal, '', kind='normal')
+
+    def __init__(self, prefix, motor_prefix, * args, **kwargs):
+        if self.__class__ is OffsetMotorBase:
+            raise TypeError('OffsetMotorBase must be subclassed with '
+                            'a "motor" component, the real motor to move.')
+        self._motor_prefix = motor_prefix
+        super().__init__(prefix, *args, **kwargs)
+
+    @pseudo_position_argument
+    def check_value(self, value):
+        return super().check_value(value)
+
+    @user_offset.sub_value
+    def _offset_changed(self, value, **kwargs):
+        """
+        The user offset was changed. Update the readback value, if possible.
+        """
+        if not hasattr(self, 'real_position'):
+            # A race condition on instantiation can cause this subscription to
+            # fire prior to the real position being available. The position
+            # will update based on this offset when available.
+            return
+
+        try:
+            # Update the internal position based on the real positioner
+            self._update_position()
+        except ophyd.utils.DisconnectedError:
+            ...
+
+    @pseudo_position_argument
+    def forward(self, pseudo_pos: tuple) -> tuple:
+        """
+        Calculate a RealPosition from a given PseudoPosition.
+
+        Parameters
+        ----------
+        pseudo_pos : PseudoPosition
+            The pseudo position input, a namedtuple.
+
+        Returns
+        -------
+        real_pos : RealPosition
+            The real position output, a namedtuple.
+        """
+        pseudo_pos = self.PseudoPosition(*pseudo_pos)
+        motor_value = pseudo_pos.pseudo_motor + self.user_offset.get()
```

Comment:
It could be that I got multiple things wrong here and they all work out together somehow.
It should probably be something like this instead:
```
def forward():
.....
motor_value = pseudo_pos.pseudo_motor - self.user_offset.get()
```

```
def inverse():
...
offset = real_pos.motor + self.user_offset.get()
```

```
def set_current_position(position):
...
new_offset = self.position[0] - position
```

I might have gotten them all reversed?

Reply:
https://github.com/pcdshub/old-hutch-configs/blob/18c7769d66a647ab558ad7ebc34d019fc98fbf48/xpp/blinst/lom.py#L446-L462

Reply:
@ZryletTC  do you think I should change this?

Reply:
No it's not something that you did. The old code has it like that as well. The issue is that this convention is the opposite of how we run most other offset values.